lib/omniauth/strategies/shopify.rb

```diff
@@ -23,6 +23,31 @@ def valid_site?
         !!(/\A(https|http)\:\/\/[a-zA-Z0-9][a-zA-Z0-9\-]*\.#{Regexp.quote(options[:myshopify_domain])}[\/]?\z/ =~ options[:client_options][:site])
       end
 
+      def valid_signature?
+        return false unless request.POST.empty?
+
+        params = request.GET
+        signature = params['hmac']
+        timestamp = params['timestamp']
+        return false unless signature && timestamp
+
+        return false unless timestamp.to_i > Time.now.to_i - 5 * 60
```

Comment:
Right, I guess app servers having slow clocks could be a concern.  However, I mostly just added this as a sanity check.  In shopify_api and shopify_python_api we actually allow timestamps as old as 24 hours, which seemed too lenient.